### DISCUSSION TITLE
Debugging random crashes & hard reboots

### DISCUSSION THREAD

**Discussion by mastoca** [YaLTeR/niri#877] [Q&A] [answered]
I have had many days of rock solid performance using niri but sometimes, the screens go black straight to a restart of the whole system. 

I'm using NIxOS and am not sure how to capture what happened and where to look to see if there's any dump that could aid in knowing what in my computer would do this.

What should I enable or where do I look for logs that could give me a clue to what is causing this.

I'm also planning to start developing and tinkering with Niri to contribute fixes or enhancements. For that case I've seen others run Niri in Niri  as a way to do testing...and that seems to be good for that case...but perhaps we should document how one would capture the logs for Niri's root instance.

**Comment by YaLTeR** [comment, marked as answer]
Well, niri certainly never reboots the whole system. In any case, you could try the last boot's journal with `journalctl -b -1`

**Reply by mastoca** [reply]
picom (which was part of my xmonad setup) was in an endless restart. Outside of that the only mention of Niri was this below...which doesn't seem to be the cause of the crashes.
```
BUG: KFENCE: use-after-free read in nv_dma_release_sgt+0x29/0x70 [nvidia]
... 
...
CPU: 3 PID: 9513 Comm: niri Tainted: P    B      O       6.6.66 #1-NixOS
``` 

I always had Wayland/Niri use the iGPU to avoid Nvidia drivers which I use only for games and ML tasks.

I'll report any further findings...but for now it looks like it was picom. @YaLTeR thank you.

**Reply by YaLTeR** [reply]
> picom (which was part of my xmonad setup) was in an endless restart.

Picom is an x11 compositor, you can't run it on wayland.

> Outside of that the only mention of Niri was this below...

That sounds like something in the nvidia driver.

**Reply by mastoca** [reply]
Yes, I had missed removing the picom service.

**Reply by mastoca** [reply]
@YaLTeR it's you were right, it is nvidia. For good measure I upgraded to kernel 6.12. I'm going to try disabling waybar for a bit to see how long it lasts.

**Reply by mastoca** [reply]
just a quick update... being a bit of a completionist here... I did a few things and the current combo changes yielded zero crashes so far today. 

What I changed:
- Kernel got updated to 6.12 (zen)
- Switched Nvidia driver to the open version AND is currently on 565.77

for NixOS (24.11) users:
```nix
  boot.kernelPackages = pkgs.linuxPackages_zen;
  hardware.nvidia = {
    modesetting.enable = true;
    open = true;
    nvidiaSettings = true;
    powerManagement.enable = false;
    powerManagement.finegrained = false;
    prime = {
      sync.enable = true;
      amdgpuBusId = ""; # enter yours see nixos nvidia wiki.
      nvidiaBusId = "" ; # see wiki.
    };
  };
```
feels good to be back in Niri all day...
